### ISSUE TITLE
Unable to find location of java.lang.Thread.setName() in debuggee JVM, for type java.lang.Thread

### ISSUE THREAD

**Issue by schipplock** [eclipse-jdt/eclipse.jdt.debug#244]
Dear JDT developers,
I can't debug my remote java application anymore.

**Here is my debug configuration in eclipse:**

<img width="174" alt="remote1" src="https://user-images.githubusercontent.com/49938/235376016-a82b6ab2-511e-49d5-a831-702245ee0fc3.png">

**I am starting my remote java application with:**

```
-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=0.0.0.0:8000
```

**I can see the typical log entry indicating the vm is picking this up:**

```
Listening for transport dt_socket at address: 8000
```

I tried to remote debug with intellij and vs code, both can connect just fine (but lack other features important to me).

**What happens when I click on "remote debug" in eclipse (1st click):**

<img width="206" alt="remote2" src="https://user-images.githubusercontent.com/49938/235376230-70746c5c-be06-43b3-876c-7abed6afdd97.png">

**I will get a workspace log:**

```
eclipse.buildId=4.27.0.20230309-1200
java.version=17.0.7
java.vendor=Eclipse Adoptium
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=de_DE
Framework arguments:  -product org.eclipse.epp.package.jee.product
Command-line arguments:  -os win32 -ws win32 -arch x86_64 -product org.eclipse.epp.package.jee.product

org.eclipse.jdt.debug
Warning
Sun Apr 30 23:06:36 CEST 2023
Unable to find location of java.lang.Thread.setName() in debuggee JVM, for type java.lang.Thread
```

**What happens when I click "remote debug" in eclipse (2nd click):**

<img width="307" alt="remote3" src="https://user-images.githubusercontent.com/49938/235376382-91715e2f-8076-408b-af24-2022a0fbb5f8.png">

**An additional workspace log entry will be created:**

```
eclipse.buildId=4.27.0.20230309-1200
java.version=17.0.7
java.vendor=Eclipse Adoptium
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=de_DE
Framework arguments:  -product org.eclipse.epp.package.jee.product
Command-line arguments:  -os win32 -ws win32 -arch x86_64 -product org.eclipse.epp.package.jee.product

org.eclipse.jdt.launching
Error
Sun Apr 30 23:09:42 CEST 2023
Failed to connect to remote VM

com.sun.jdi.connect.spi.ClosedConnectionException
	at org.eclipse.jdi.internal.connect.SocketTransportService.readHandshake(SocketTransportService.java:225)
	at org.eclipse.jdi.internal.connect.SocketTransportService$3.run(SocketTransportService.java:185)
	at java.base/java.lang.Thread.run(Thread.java:833)

```

I'm using Eclipse IDE 2023-03 (4.27) for Enterprise Java and Web.
JDT Version: Version: 3.19.0.v20230302-0300, Build id: I20230302-0300.

**My eclipse.ini is unmodified. Eclipse is using the following vm to start:**

```
-vm
plugins/org.eclipse.justj.openjdk.hotspot.jre.full.win32.x86_64_17.0.7.v20230425-1502/jre/bin
```

**My list of installed JREs:**

<img width="504" alt="remote4" src="https://user-images.githubusercontent.com/49938/235376785-8e063ab1-8de5-4ab4-a5ac-3a62adde2576.png">

I'm running Windows 11 Pro and using Oracles JDK 11.0.18.
My remote java application is running Ubuntu 22.04, Tomcat 10.1.8 and Temurin JRE 11.0.19_7 (OpenJDK11U-jre_x64_linux_hotspot_11.0.19_7.tar.gz). The remotely deployed web application is compiled using the Oracle JDK 11 on Windows.

I probably missed something obvious, but I can't see it :(.
I'm here to kindly ask for a hint. I'm clueless now :).

Thanks in advance,
Andreas

**Comment by iloveeclipse** [comment]
Just a guess: try to start your Eclipse **not** with justj **jre** but with a **full** JDK (doesn't matter if from Oracle or Adoptium).

**Comment by schipplock** [comment]
Hi Andrey,
thank you for taking time helping me.

I removed everything Oracle from my harddrive and installed Temurin 17 JDK, updated to the latest maven (3.9.1) and I'm using the latest maven compiler plugin now (3.11.0). I then replaced Temurin 11 with Temurin 17 in my Tomcat Docker Image and recompiled my project with Java 17.

I am now able to remote debug again :)!

Thank you very much.

Regards,
Andreas
